### DISCUSSION TITLE
Oh-my-zsh in `.gitconfig` template

### DISCUSSION THREAD

**Discussion by claughinghouse** [twpayne/chezmoi#3970] [Q&A] [answered]
I manage my `.gitconfig` with chezmoi but recently started running into an issue where chezmoi tries to overwrite the key that `omz` sets in the gitconfig:

```
[oh-my-zsh]
	git-commit-alias = xxyyzzlongsha
```

I tried writing a template to just ignore the line since chezmoi currently supports excluding directories, not lines in files this way:

```
{{ if eq true false }}
[oh-my-zsh]
	git-commit-alias = xxyyzzlongsha
{{ end }}

```

But each time I tun `omz update` the sha changes and chezmoi catches it on next update. Is there a elegant way to ignore whatever the value of `git-commit-alias` within chezmoi?

**Comment by bradenhilton** [comment]
If you can't prevent Oh My Zsh from writing to `.gitconfig`, you may need to use a [`modify_` script or template](https://www.chezmoi.io/user-guide/manage-different-types-of-file/#manage-part-but-not-all-of-a-file).

[Handle configuration files which are externally modified](https://www.chezmoi.io/user-guide/manage-different-types-of-file/#handle-configuration-files-which-are-externally-modified) may also be useful.

**Comment by twpayne** [comment, marked as answer]
Another, admittedly hacky, option is to make `.gitconfig` a template and extract the value written by Oh My Zsh. Something like:

```
[oh-my-zsh]
    git-commit-alias = {{ index (output "grep" "git-commit-alias" (joinPath .chezmoi.homeDir ".gitconfig") | splitList "=") 1 | trim }}
```

This is *really* ugly though.

**Reply by claughinghouse** [reply]
This is similar to what I was able to come up with (mine is also abeit ugly) since my `gitconfig` was already a template:

```
{{- $homeDir := .chezmoi.homeDir }}
{{- $zshDir := printf "%s/.oh-my-zsh" $homeDir }}
{{ $gitHash := output "git" "-C" $zshDir "rev-parse" "HEAD" | trim }}

[oh-my-zsh]
  git-commit-alias = {{ $gitHash }}

```
